tests/spin_targets.py

```diff
@@ -0,0 +1,108 @@
+            print(f'{method_name}: creating {dirname}...')
+            target.makedirs(dirname)
+
+            try:
+                for key, value in data.items():
+                    path = os.path.join(dirname, key)
+                    print(f'{method_name}: writing {value!r} to {path}...')
+                    target.write_value(path, value, verify=False,
+                                       as_root=target.conn.connected_as_root)
+
+                print(f'{method_name}: reading values from target...')
+                raw_result = target.read_tree_values_flat(dirname)
+                result = {os.path.basename(k): v for k, v in raw_result.items()}
+            except Exception as ex:
+                raise TargetStableError('Error while processing the tree!') from ex
+            finally:
+                print(f'{method_name}: removing {dirname}...')
+                target.remove(dirname)
+
+            self.assertEqual({k: v.strip()
+                              for k, v in data.items()},
+                             result)
+
+            print(f'{method_name}: removing {target.working_directory}...')
+            target.remove(target.working_directory)
+
+        def create_targets():
+            targets = []
+
+            a_target = AndroidTarget(connection_settings={'device': 'emulator-5554'},
+                                     working_directory='/data/local/tmp/devlib-target')
+            targets.append(a_target)
+
+            l_target = LinuxTarget(connection_settings={'host': 'example.com',
+                                                        'username': 'username',
+                                                        'password': 'password'},
+                                   working_directory='/tmp/devlib-target')
+            targets.append(l_target)
+
+            ll_target = LocalLinuxTarget(connection_settings={'unrooted': True},
+                                         working_directory='/tmp/devlib-target')
+            targets.append(ll_target)
+
+            qemu_runner = QEMUTargetRunner(qemu_params={
+                                               'kernel_image': '/home/username/devlib/tools/buildroot/buildroot-v2023.11.1/output/images/Image'},
+                                           connection_settings={'host': '127.0.0.1',
+                                                                'port': 8022,
+                                                                'username': 'root',
+                                                                'password': 'root',
+                                                                'strict_host_check': False},
+                                           target_params={'working_directory': '/tmp/devlib-target'})
+            targets.append(qemu_runner.target)
+
+            return targets
+
+        method_name = self.__class__.test_read_multiline_values.__qualname__
+
+        targets = create_targets()
+        for target in targets:
```

Comment:
it's probably good enough for now but when we write more tests, we probably don't want to create a new set of target every time, so we will want to use the unittest infrastructure to keep those alive. Or maybe not if it stinks too much, let's see later